crates/matrix-sdk/src/http_client/mod.rs

```diff
@@ -48,16 +50,47 @@ pub(crate) use native::HttpSettings;
 
 pub(crate) const DEFAULT_REQUEST_TIMEOUT: Duration = Duration::from_secs(30);
 
+#[derive(Clone, Debug)]
+struct MaybeSemaphore(Arc<Option<Semaphore>>);
+
+#[allow(dead_code)] // holding this until drop is all we are doing
+struct MaybeSemaphorePermit<'a>(Option<SemaphorePermit<'a>>);
+
+impl MaybeSemaphore {
+    fn new(max: Option<NonZeroUsize>) -> Self {
+        let inner = max.map(|i| Semaphore::new(i.into()));
+        MaybeSemaphore(Arc::new(inner))
+    }
+
+    async fn acquire(&self) -> MaybeSemaphorePermit<'_> {
+        match self.0.as_ref() {
+            Some(inner) => {
+                // ignoring errors as we never close this
```

Comment:
What kind of errors do you mean here?

Reply:
> If the semaphore has been closed, this returns an [AcquireError](https://docs.rs/tokio/latest/tokio/sync/struct.AcquireError.html). 

This only ever happens when there the Semphore was explicitly closed, which this `MaybeSemaphore` doesn't ever do. So the error can never occur and can be ignored.

Reply:
Can you please update the comment to make this clear?